### ISSUE TITLE
Not able to remove server from cluster with name provided during add operation

### ISSUE THREAD

**Issue by hemanthnakkina** [canonical/microceph#125]
Added a server to the cluster with name microceph/14 and not able to use the same name to remove the server.

Error message:
No remote exists with the given name "microceph%2F14"

Expectation is to be able to use the same <NAME> provided during add operation for delete operation as well.
If the basic assumption is <NAME> should be hostname, it should spell out explicitly in documentation and help.

Note: ``microceph cluster list`` shows the servers with hostname instead of name provided during add operation.

Steps to reproduce:
``
sudo microceph cluster add microceph/14
sudo microceph cluster join <token>
sudo microceph cluster remove microceph/14
``

**Comment by tomponline** [comment]
@masnax @stgraber sounds like this may be a microcloud issue with not using `net/url.PathUnescape()` on the URL segment.

**Comment by tomponline** [comment]
LXD uses this along with `mux.UseEncodedPath()` so that gorilla isn't tempted to interpret these as URL segments.

**Comment by stgraber** [comment]
Right, we can do that for specific routes that do need to include slashes but that's pretty rare.

In this instance, the problem is the slash in the name which we should have allowed in the first place.
